test/integration/scheduler/predicates_test.go

```diff
@@ -1041,18 +1042,45 @@ var (
 func TestUnschedulablePodBecomesSchedulable(t *testing.T) {
 	tests := []struct {
 		name   string
-		init   func(kubernetes.Interface) error
+		init   func(kubernetes.Interface, string) error
```

Comment:
This new parameter is odd. What if we need more nodes for one, or if we want to update pods instead of nodes.

Reply:
Yes, definitely we will need to add more params. For instance, I am thinking to add the `namespace` in the future to be able to create/update pods in `init`/`update`.

I don't have strong opinion about the node name. I suspect we won't need more than one node for future tests and I thought passing it as a parameter is more explicit than modifying a shared variable.

Reply:
Unless you have a new solution. I would propose to keep it this way for now and see how the new tests evolve. If `init` and `update` signature become too large we can easily come back to previous design.

WDYT?

Reply:
to keep it simple, I prefer we remove it, since we don't need it.

Reply:
What's your proposal? Would you rather have the shared global variable?

Reply:
hardcode the name. This is expected in tests. It makes the test easier to read.

Reply:
a couple of points:
- don't hardcode the node name to something simple as `node-1` as this node can exist in the cluster already and nodes are global.
- if the test cases one day run in parallel using a shared node will fail (currently they run in serial).

Reply:
I just said that these tests will never run in parallel, because one of the tests expects no nodes at all at the beginning.

Reply:
but having a more descriptive node name might be handy.

Reply:
Added, thanks!